### PR TITLE
Meilleure stabilité des tests e2e

### PR DESCRIPTION
**Pull request by fabienheureux**
# Description succincte du problème résolu

problème de tests e2e instables 

**🗺️ contexte**: les tests fonctionnent pas pareil chez Nico et moi (je viens de comprendre pourquoi !!)

**💡 quoi**:
- correctifs sur deux tests un peu flaky
- ajout du build dans le setup de playwright en local pour ne plus avoir à build manuellement avec localhost:8888 comme base_url

**🎯 pourquoi**: 

- car le setup ne tenait pas compte de certains changements faits il y a longtemps

**🤔 comment**: 

le diff est plutôt explicite :/ 

**🤓 comment tester**:

Prérequis sur la machine du développeur pour lancer les tests e2e en local :

```bash
# 1. depuis la racine du repo : démarrer Postgres
docker compose --profile lvao up -d

# 2. depuis webapp/ : (re)crée la base webapp_sample, migre, build le JS
make prepare-e2e-test-local

# 3. lance les tests
make e2e-test
```

`prepare-e2e-test-local` recrée systématiquement `webapp_sample` (drop + create) pour
repartir d'un état propre à chaque run, via `manage.py create_webapp_sample_db`. La
variable d'env `DB_WEBAPP_SAMPLE` (déjà dans `.env.template`) pointe vers cette base.

Voir [docs/reference/webapp/README.md](../blob/main/docs/reference/webapp/README.md#running-e2e-tests-locally)
pour le détail.

## Exemple résultats / UI / Data

image

## Auto-review

Les trucs à faire avant de demander une review :

- [ ] J'ai bien relu mon code
- [ ] La CI passe bien
- [ ] En cas d'ajout de variable d'environnement, j'ai bien mis à jour le `.env.template`
- [ ] J'ai ajouté des tests qui couvrent le nouveau code
- [ ] En cas d'ajout de dépendance, j'ai bien respecté un cooldown de 7 jours pour éviter les [Supply Chain Attacks](https://cheatsheetseries.owasp.org/cheatsheets/Software_Supply_Chain_Security_Cheat_Sheet.html)

## ✅ Reste à faire (PR en cours)

- [ ] Commiter le fix du chemin SQL dans `create_webapp_sample_db` (pointait vers `webapp/scripts/sql/` au lieu de `scripts/sql/`)
- [ ] Commiter le passage de la migration `qfdmd.0045` en no-op (le texte search config `wagtail_french` est désormais créé une seule fois dans `scripts/sql/create_extensions.sql`)

## 📆 A faire (prochaine PR)

- [ ]

### CODE REVIEW

File: .github/workflows/_e2e.yml [hunk 1]
```diff
@@ -95,6 +96,10 @@ jobs:
           database-url: ${{ env.DATABASE_URL }}
           dump-path: database-dump.pgdump
 
+      - name: Fix duplicate MainMenu records
+        working-directory: webapp
+        run: uv run python manage.py fix_duplicate_mainmenu
+
```
Comment (kolok) [inline .github/workflows/_e2e.yml]: Est-ce qu'on n'a pas intérêt à passer cette commande avant le test qui fail ? comme un hook before ?
Ou alors que le test soit résiliant
si ça foire avec plusieurs MainMenu, est-ce que c'est pas un signe qu'on devrait interdire plusieurs MainMenu
bref, je te propose qu'on en echange pour plus de détails

Comment (fabienheureux) [reply .github/workflows/_e2e.yml]: c'est pas un problème du test, c'est la manière dont on génère la db sample le soucis, on en parle tout à l'heure

File: .github/workflows/_e2e.yml [hunk 2]
```diff
@@ -83,6 +83,7 @@ jobs:
         uses: ./.github/actions/setup-e2e-tools
         with:
           build-frontend: "true"
+          base-url: ${{ inputs.base-url || 'http://localhost:8888' }}
```
Comment (fabienheureux) [inline .github/workflows/_e2e.yml:86]: dans certains edge case, le build parcel comportait l'url de prod pour l'iframe, ce qui fait qu'on testait pas bien

File: webapp/Makefile [hunk 3]
```diff
@@ -147,7 +147,6 @@ prepare-e2e-test:
 	$(DJANGO_ADMIN) enable_trigram
 	make rebuild-search-index
 	make createsuperuser-example
-	BASE_URL=http://localhost:8888 npm run build
```
Comment (fabienheureux) [inline webapp/Makefile:150]: je bascule dans la config playwright directement

File: webapp/e2e_tests/search.spec.ts [hunk 4]
```diff
@@ -249,34 +249,15 @@ test.describe("Positionnement de l'autocomplete en hauteur contrainte", () => {
...
-    expect(measurements!.rectBottom).toBeLessThanOrEqual(measurements!.bodyHeight)
-    // Tolerance = clamp margin, not flakiness slop.
-    expect(measurements!.bodyHeight - measurements!.rectBottom).toBeLessThanOrEqual(10)
-    expect(measurements!.scrollHeight).toBeGreaterThan(measurements!.clientHeight)
-    expect(measurements!.overflowY).toBe("auto")
-    // @iframe-resizer/child skips elements with this attribute when sizing the iframe.
-    expect(measurements!.hasIframeIgnore).toBe(true)
+    const listbox = page.locator(SEARCH_RESULTS_DROPDOWN_SELECTOR).getByRole("listbox")
+    const box = await listbox.boundingBox()
+    expect(box?.height! + box?.y! < pageHeight)
```
Comment (fabienheureux) [inline webapp/e2e_tests/search.spec.ts]: test tout nul généré par un llm un jour je pense, j'ai revu en l'écrivant avec la doc de playwright, bah c'est vachement mieux hehe

File: .github/workflows/_e2e.yml [hunk 5]
```diff
@@ -34,6 +34,8 @@ jobs:
   test:
     name: Playwright (${{ inputs.ref }})
     runs-on: ubuntu-latest
+    env:
+      BASE_URL: ${{ inputs.base-url }}
```
Comment (fabienheureux) [inline .github/workflows/_e2e.yml]: centralisation du BASE_URL pour ne pas avoir à le répéter à droite à gauche, on le définit une fois et on le réutilise partout

File: .github/workflows/e2e.yml [hunk 6]
```diff
@@ -58,7 +58,7 @@ jobs:
       - name: Prepare Django database
         uses: ./.github/actions/prepare-django-db
         with:
-          cache-key: db-dump-${{ hashFiles('**/migrations/**/*.py') }}-${{ steps.date.outputs.value }}
+          cache-key: ${{ github.run_id }}-db-dump-${{ hashFiles('**/migrations/**/*.py') }}-${{ steps.date.outputs.value }}
```
Comment (fabienheureux) [inline .github/workflows/e2e.yml]: je cache la db sample qu'à l'échelle du run github actions pour éviter d'avoir des trucs décalé avec le changement en cours, on peut le merger mais ce sera à peaufiner un jour

File: webapp/e2e_tests/helpers.ts [hunk 7]
```diff
@@ -874,7 +875,7 @@ export async function typeSearchQuery(
   const searchInput = locator.locator(SEARCH_INPUT_SELECTOR)
   await searchInput.click()
   await searchInput.fill("")
-  await searchInput.pressSequentially(query, { delay: 50 })
+  await searchInput.pressSequentially(query, { delay: 200 })
```
Comment (fabienheureux) [inline webapp/e2e_tests/helpers.ts]: j'allonge ici parce que ca rendait le test flaky d'avoir trop peu d'intervalle entre chaque keypress

File: webapp/qfdmo/management/commands/create_webapp_sample_db.py [hunk 8]
```diff
@@ -29,25 +29,31 @@ class Command(BaseCommand):
...
-            self.stdout.write(
-                self.style.WARNING("La base de données webapp_sample existe déjà")
+        # Always start from a clean database to avoid leftover state
+        # (e.g. partial migrations) from previous runs. Terminate other
+        # backends first since DROP DATABASE fails while connections are open.
+        with conn.cursor() as cursor:
+            cursor.execute(
+                "SELECT pg_terminate_backend(pid) FROM pg_stat_activity "
+                "WHERE datname = 'webapp_sample' AND pid <> pg_backend_pid();"
             )
```
Comment (kolok) [inline webapp/qfdmo/management/commands/create_webapp_sample_db.py:42]: A quoi sert ce select ?

File: docs/reference/webapp/README.md
Comment (kolok) [inline docs/reference/webapp/README.md:1]: Note pour plus tard,on devrait bouger la partie tests dans sont propre sous .md : `tests.md`

**Review by kolok — APPROVED** [review]

File: docs/reference/webapp/README.md
Comment (fabienheureux) [reply docs/reference/webapp/README.md:1]: aligné

File: webapp/qfdmo/management/commands/create_webapp_sample_db.py (hunk 8 above)
Comment (fabienheureux) [reply webapp/qfdmo/management/commands/create_webapp_sample_db.py:42]: ca m'arrivait d'avoir un playwright mal quitté toujours connecté au serveur de db, là ca récupère les connexions actives à webapp_sample et les kill avant de drop la db (on peut pas drop une db avec des connexion active)
